src/aap_eda/api/filters/eda_credential.py

```diff
@@ -24,7 +24,7 @@ class EdaCredentialFilter(django_filters.FilterSet):
         label="Filter by EDA credential name.",
     )
     credential_type_id = django_filters.NumberFilter(
-        field_name="credential_type_id",
+        field_name="credential_type__id",
```

Comment:
@Dostonbek1 @hsong-rh Why does this have to be double underscore? This was added in PR https://github.com/ansible/eda-server/pull/819

Reply:
@mkanoor I feel both ways should be ok. One is on the field directly, the other is based on the relation.

Reply:
This might be overlook on my end from resolving merge conflicts. I'll test it and push a fix in a bit.

Reply:
It seems to work fine in both scenarios but put the `credential_type_id` back so as not to divert from main branch changes.